FIX: _monitor was packing configuration wrong.

danielballan opened the pull request:
Stop-stopping bug in the RunEngine monitoring code organized the configuration into the descriptor with the wrong layout. The currently-deployed code acquired data without complaint but the databroker errors (rightly) when it can't find the keys it expects.

In addition to this PR, which should probably be merged and deployed ASAP, we should:

* add more tests that exercise configuration (I will add at least one to this PR later today, but I have to teach first.)
* make the event_model stricter (mentioned way back in https://github.com/NSLS-II/bluesky/issues/134) -- this should have produced an error at acquisition time!

Can you look into this next, @licode? Reference for how the Event Descriptor is organized: http://nsls-ii.github.io/bluesky/event_descriptors.html

tacaswell commented on bluesky/run_engine.py:
```diff
@@ -1258,9 +1258,11 @@ def _monitor(self, msg):
+        config[obj.name]['data_keys'] = obj.describe_configuration()
+        for key, val in list(obj.read_configuration().items()):
+            config[obj.name]['data'][key] = sanitize_np(val['value'])
+            config[obj.name]['data'][key] = val['timestamp']
```
I think the second one should be `'timestamps'`?

danielballan replied:
No, the structure of `read()` and `read_configuration()` is `{<data_key>: {'value': ..., 'timestamp': ...}, ...}`

danielballan replied:
:sheep:.

danielballan commented:
Addressed line comment and added test. Anaconda.org connectivity is killing our Travis builds more than half the time today, so odds are Travis will fail. Tests pass locally.

licode commented:
Busy schedule today. Sure, I will catch up.

tacaswell commented on bluesky/run_engine.py:
```diff
@@ -1258,9 +1258,11 @@ def _monitor(self, msg):
-            val['value'] = sanitize_np(val['value'])
+        config = {obj.name: {'data': {}, 'timestamps': {}}}
+        config[obj.name]['data_keys'] = obj.describe_configuration()
+        for key, val in list(obj.read_configuration().items()):
```
minor :bike: :house_with_garden: , we do not need the `list` here.

danielballan replied:
Rookie move there... what was I thinking?
